doc/source/user/tutorial-ma.rst

```diff
@@ -0,0 +1,377 @@
+function, making sure we select only the columns with actual numbers instead of
+the first three columns which contain location data. We also skip the first 7
+rows of this file, since they contain other data we are not interested in.
+Separately, we will extract the information about dates and location for this
+data. 
+
+.. ipython:: python
+
+   # Note we are using skip_header and usecols to read only portions of the data
+   # file into each variable.
+   # Read just the dates for columns 3-7 from the first row
+   dates = np.genfromtxt(filename, dtype=np.unicode_, delimiter=",",
+	                 max_rows=1, usecols=range(3, 17),
+			 encoding="utf-8-sig")
+   # Read the names text names of the geographic locations from the first two
+   # columns, skipping the first seven rows
+   locations = np.genfromtxt(filename, dtype=np.unicode_, delimiter=",",
+	                     skip_header=7, usecols=(0, 1),
+			     encoding="utf-8-sig")
+   # Read the numeric data from just the first 14 days
+   nbcases = np.genfromtxt(filename, dtype=np.int_, delimiter=",",
+                           skip_header=7, usecols=range(3, 17),
+			   encoding="utf-8-sig")
+
+Included in the :func:`numpy.genfromtxt` function call, we have selected the
+:class:`numpy.dtype` for each subset of the data (either an integer -
+:class:`numpy.int_` - or a string of characters - :class:`numpy.unicode_`). We
+have also used the ``encoding`` argument to select ``utf-8-sig`` as the encoding
+for the file (read more about encoding in the `official Python documentation
+<https://docs.python.org/3/library/codecs.html#encodings-and-unicode>`__). You
+can read more about the :func:`numpy.genfromtxt` function from
+the :func:`Reference Documentation <numpy.genfromtxt>` or from the
+:doc:`Basic IO tutorial <basics.io.genfromtxt>`.
+
+**Exploring the data**
+
+First of all, we can plot the whole set of data we have and see what it looks
+like. In order to get a readable plot, we select only a few of the dates to
+show in our :func:`x-axis ticks <matplotlib.pyplot.xticks>`. Note also that in
+our plot command, we use ``nbcases.T`` (the transpose of the ``nbcases`` array)
+since this means we will plot each row of the file as a separate line. We choose
+to plot a dashed line (using the ``'--'`` line style). See the
+`matplotlib <https://matplotlib.org/>`_ documentation for more info on this.
+
+.. ipython:: python
+
+    import matplotlib.pyplot as plt
+    selected_dates = [0, 3, 11, 13]
+    plt.plot(dates, nbcases.T, '--');
+    plt.xticks(selected_dates, dates[selected_dates]);
+    @savefig plot_covid_1.png
+    plt.title("COVID-19 cumulative cases from Jan 21 to Feb 3 2020");
+    
+.. note::
+
+   If you are executing the commands above in the IPython shell, it might be
+   necessary to use the command ``plt.show()`` to show the image window. Note
+   also that we use a semicolon at the end of a line to suppress its output, but
+   this is optional.
```

Comment:
One more option would be to use the `jupyter-execute` directive from `jupyter_sphinx`, which works in almost the same way as `ipython`, but also lets you download a complete script at the end.

Reply:
Nice to know, I'll leave this one as is for now but I'll definitely have a look at that.